spec/rubocop/cop/rails/reflection_class_name_spec.rb

```diff
@@ -56,4 +56,20 @@
       belongs_to :account, class_name: :Account
     RUBY
   end
+
+  it 'does not register an offense when using self.name for `class_name`' do
+    expect_no_offenses(<<~RUBY)
+      has_many :accounts, class_name: self.name, foreign_key: :account_id
```

Comment:
Won't this trigger the `RedundantSelf` cop?

Reply:
I'm not sure actually. Does that rule apply at the class level too? If the PR still makes sense I can update this.

Reply:
Yes, it detects it on the class level, too.

>  `self.name` reads more clearly in this class context

Fair enough. And the cop tolerates the following:
```ruby
has_many :accounts, class_name: name
```
So everything is fine 👍